flashlight/fl/tensor/backend/jit/ir/IndexNode.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "flashlight/fl/tensor/backend/jit/ir/IndexNode.h"
+
+#include <stdexcept>
+
+#include "flashlight/fl/tensor/backend/jit/JitTensorBase.h"
+
+namespace fl {
+
+namespace {
+
+Dim ceilDiv(Dim numerator, Dim denominator) {
+  return (numerator + denominator - 1) / denominator;
+}
+
+// TODO refactor with common index logic in OneDnnTensor; maybe put things like
```

Comment:
+1. The ArrayFireBackend also has an implementation of this function which is probably less correct, and this is useful for testing too

Reply:
Updated the TODO to mention AF as well.